backend/yubikey.go

```diff
@@ -0,0 +1,218 @@
+package backend
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/md5"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"time"
+
+	"github.com/foxboron/sbctl/config"
+	"github.com/foxboron/sbctl/hierarchy"
+	"github.com/foxboron/sbctl/logging"
+
+	"github.com/go-piv/piv-go/v2/piv"
+)
+
+type YubikeyData struct {
+	Algorithm   piv.Algorithm   `json:"algorithm"`
+	PinPolicy   piv.PINPolicy   `json:"pinPolicy"`
+	TouchPolicy piv.TouchPolicy `json:"touchPolicy"`
+	Slot        string          `json:"slot"`
+	PublicKey   string          `json:"publicKey"`
+}
+
+type Yubikey struct {
+	keytype       BackendType
+	cert          *x509.Certificate
+	yubikeyReader *config.YubikeyReader
+	algorithm     piv.Algorithm
+	pinPolicy     piv.PINPolicy
+	touchPolicy   piv.TouchPolicy
+}
+
+func NewYubikeyKey(yubikeyReader *config.YubikeyReader, hier hierarchy.Hierarchy) (*Yubikey, error) {
+	cert, err := yubikeyReader.GetPIVKeyCert()
+	if err != nil {
+		if !errors.Is(err, piv.ErrNotFound) {
+			return nil, err
```

Comment:
Lets wrap this error!
```suggestion
			return nil, fmt.Errorf("failed finding yubikey: %v", err)
```

Reply:
Should use `%w` not `%v` when wrapping errors

Reply:
@lespea only if you want the caller to depend on the error "type". And in this case we don't know yet.